os/board/common/partitions.c

```diff
@@ -78,40 +80,48 @@ FAR struct mtd_dev_s *mtd_initialize(void)
 	return mtd;
 }
 
-static void type_specific_initialize(int minor, FAR struct mtd_dev_s *mtd_part, int partno, const char *types)
+static void type_specific_initialize(int minor, FAR struct mtd_dev_s *mtd_part, int partno, const char *types, partition_info_t *partinfo)
 {
+	if (partinfo == NULL) {
+		lldbg("ERROR: partinfo is NULL\n");
+		return;
+	}
 #ifdef CONFIG_MTD_FTL
-		if (!strncmp(types, "ftl,", 4)
+	if (!strncmp(types, "ftl,", 4)
 #ifdef CONFIG_FS_ROMFS
-		|| !strncmp(types, "romfs,", 6)
+	|| !strncmp(types, "romfs,", 6)
 #endif
 #ifdef CONFIG_BINARY_MANAGER
-		|| !strncmp(types, "kernel,", 7)
+	|| !strncmp(types, "kernel,", 7)
 #endif
-		) {
-			if (ftl_initialize(partno, mtd_part)) {
-				lldbg("ERROR: failed to initialise mtd ftl errno :%d\n", errno);
-				return;
-			}
-		} else
+	) {
+		if (ftl_initialize(partno, mtd_part)) {
+			lldbg("ERROR: failed to initialise mtd ftl errno :%d\n", errno);
+			return;
+		} else {		
+			partinfo->romfs_partno = partno;
```

Comment:
This line is NOT only for romfs. This will cause overwrite issue.

Reply:
Done. Thanks.